x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -548,61 +548,100 @@ public void checkPrivileges(
                 listener.onResponse(result);
                 return;
             }
+            maybeCachingListener = listener.delegateFailure((delegateListener, privilegesCheckResult) -> {
+                try {
+                    simpleRole.cacheHasPrivileges(settings, privilegesToCheck, privilegesCheckResult);
+                } catch (ExecutionException e) {
```

Comment:
I think we should be more liberal here and catch `Exception` instead of just `ExecutionException`. We don't want any future possible error to slip through this catch block.